### PR TITLE
PR for task ADO #4454

### PR DESCRIPTION
**Pull request by bquantump**
PR for https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/4454

### CODE REVIEW

File: azure-proto-network/NetworkInterfaceContainer.cs [hunk 1]
```diff
@@ -65,21 +67,37 @@ public async override Task<ArmOperation<NetworkInterfaceOperations>> StartCreate
 
             return new ArmBuilder<NetworkInterfaceOperations, PhNetworkInterface>(this, new PhNetworkInterface(nic));
         }
+    
+        public Pageable<NetworkInterfaceOperations> List(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/NetworkInterfaceContainer.cs:71]: Can we move the { to the newline?

File: client/Scenarios/ListAllResourceGroupsByName.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+using azure_proto_compute;
```
Comment (m-nash) [inline client/Scenarios/ListAllResourceGroupsByName.cs:1]: Is this file needed any longer?  Since listing resource groups by name was deemed not possible in azure API?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: client/Scenarios/ListAllResourceGroupsByName.cs (hunk 2 above)
Comment (bquantump) [reply client/Scenarios/ListAllResourceGroupsByName.cs:1]: Nope, can remove

File: azure-proto-network/NetworkInterfaceContainer.cs (hunk 1 above)
Comment (bquantump) [reply azure-proto-network/NetworkInterfaceContainer.cs]: yup

File: azure-proto-network/NetworkInterfaceContainer.cs [hunk 3]
```diff
@@ -65,21 +67,37 @@ public async override Task<ArmOperation<NetworkInterfaceOperations>> StartCreate
...
         }
+    
+        public Pageable<NetworkInterfaceOperations> List(CancellationToken cancellationToken = default)
+        {
+            return new PhWrappingPageable<NetworkInterface, NetworkInterfaceOperations>(
+                Operations.List(Id.Name, cancellationToken),
+                this.convertor());
+        }
+
+        public AsyncPageable<NetworkInterfaceOperations> ListAsync(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/NetworkInterfaceContainer.cs:78]: still need to move to newline

File: azure-proto-network/NetworkInterfaceContainer.cs [hunk 4]
```diff
@@ -65,21 +67,37 @@ public async override Task<ArmOperation<NetworkInterfaceOperations>> StartCreate
...
 
-        public AsyncPageable<NetworkInterfaceOperations> ListAsync(ArmSubstringFilter filter = null, int? top = null, CancellationToken cancellationToken = default)
+        public AsyncPageable<ArmResourceOperations> ListByNameAsync(ArmSubstringFilter filter, int? top = null, CancellationToken cancellationToken = default)
         {
             ArmFilterCollection filters = new ArmFilterCollection(PhNetworkInterface.ResourceType);
             filters.SubstringFilter = filter;
-            return ResourceListOperations.ListAtContextAsync<NetworkInterfaceOperations, PhNetworkInterface>(ClientContext, Id, filters, top, cancellationToken);
+            return ResourceListOperations.ListAtContextAsync<ArmResourceOperations, ArmResource>(ClientContext, Id, filters, top, cancellationToken);
+        }
+        private  Func<NetworkInterface, NetworkInterfaceOperations> convertor(){
```
Comment (m-nash) [inline azure-proto-network/NetworkInterfaceContainer.cs:98]: still need to move to newline

File: azure-proto-network/NetworkSecurityGroupContainer.cs [hunk 5]
```diff
@@ -86,20 +88,34 @@ public async override Task<ArmOperation<NetworkSecurityGroupOperations>> StartCr
 
             return new ArmBuilder<NetworkSecurityGroupOperations, PhNetworkSecurityGroup>(this, new PhNetworkSecurityGroup(nsg));
         }
+        public Pageable<NetworkSecurityGroupOperations> List(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/NetworkSecurityGroupContainer.cs:91]: still need to move to newline

File: azure-proto-network/NetworkSecurityGroupContainer.cs [hunk 6]
```diff
@@ -86,20 +88,34 @@ public async override Task<ArmOperation<NetworkSecurityGroupOperations>> StartCr
...
+            return new PhWrappingPageable<NetworkSecurityGroup, NetworkSecurityGroupOperations>(
+                Operations.List(Id.Name, cancellationToken),
+                this.convertor());
+        }
+        public AsyncPageable<NetworkSecurityGroupOperations> ListAsync(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/NetworkSecurityGroupContainer.cs:96]: still need to move to newline

File: azure-proto-network/PublicIpAddressContainer.cs [hunk 7]
```diff
@@ -56,18 +58,33 @@ public async override Task<ArmOperation<PublicIpAddressOperations>> StartCreateA
             return new ArmBuilder<PublicIpAddressOperations, PhPublicIPAddress>(this, new PhPublicIPAddress(ipAddress));
         }
 
-        public Pageable<PublicIpAddressOperations> List(ArmSubstringFilter filter = null, int? top = null, CancellationToken cancellationToken = default)
+        public Pageable<PublicIpAddressOperations> List(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/PublicIpAddressContainer.cs:61]: still need to move to newline

File: azure-proto-network/PublicIpAddressContainer.cs [hunk 8]
```diff
@@ -56,18 +58,33 @@ public async override Task<ArmOperation<PublicIpAddressOperations>> StartCreateA
...
+            return new PhWrappingPageable<PublicIPAddress, PublicIpAddressOperations>(
+                Operations.List(Id.Name, cancellationToken),
+                this.convertor());
+        }
+
+        public AsyncPageable<PublicIpAddressOperations> ListAsync(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/PublicIpAddressContainer.cs:67]: still need to move to newline

File: azure-proto-network/PublicIpAddressContainer.cs [hunk 9]
```diff
@@ -56,18 +58,33 @@ public async override Task<ArmOperation<PublicIpAddressOperations>> StartCreateA
...
 
-        public AsyncPageable<PublicIpAddressOperations> ListAsync(ArmSubstringFilter filter = null, int? top = null, CancellationToken cancellationToken = default)
+        public AsyncPageable<ArmResourceOperations> ListByNameAsync(ArmSubstringFilter filter, int? top = null, CancellationToken cancellationToken = default)
         {
             ArmFilterCollection filters = new ArmFilterCollection(PhPublicIPAddress.ResourceType);
             filters.SubstringFilter = filter;
-            return ResourceListOperations.ListAtContextAsync<PublicIpAddressOperations, PhPublicIPAddress>(ClientContext, Id, filters, top, cancellationToken);
+            return ResourceListOperations.ListAtContextAsync<ArmResourceOperations, ArmResource>(ClientContext, Id, filters, top, cancellationToken);
+        }
+        private Func<PublicIPAddress, PublicIpAddressOperations> convertor(){
```
Comment (m-nash) [inline azure-proto-network/PublicIpAddressContainer.cs:86]: still need to move to newline

File: azure-proto-network/SubnetContainer.cs [hunk 10]
```diff
@@ -63,20 +64,21 @@ public async override Task<ArmOperation<SubnetOperations>> StartCreateAsync(stri
             return new ArmBuilder<SubnetOperations, PhSubnet>(this, new PhSubnet(subnet, location ?? DefaultLocation));
         }
 
-        public Pageable<SubnetOperations> List(CancellationToken cancellationToken = default)
-        {
-            var subnetClient = GetClient((uri, cred) => new NetworkManagementClient(Id.Subscription, uri, cred)).Subnets;
+        public Pageable<SubnetOperations> List(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/SubnetContainer.cs:67]: still need to move to newline

File: azure-proto-network/SubnetContainer.cs [hunk 11]
```diff
@@ -63,20 +64,21 @@ public async override Task<ArmOperation<SubnetOperations>> StartCreateAsync(stri
...
-                subnetClient.List(Id.ResourceGroup, Id.Name, cancellationToken),
-                s => new SubnetOperations(ClientContext, new PhSubnet(s, DefaultLocation)));
+                Operations.List(Id.ResourceGroup, Id.Name, cancellationToken),
+                this.convertor());
         }
 
-        public AsyncPageable<SubnetOperations> ListAsync(CancellationToken cancellationToken = default)
-        {
-            var subnetClient = GetClient((uri, cred) => new NetworkManagementClient(Id.Subscription, uri, cred)).Subnets;
+        public AsyncPageable<SubnetOperations> ListAsync(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/SubnetContainer.cs:73]: still need to move to newline

File: azure-proto-network/SubnetContainer.cs [hunk 12]
```diff
@@ -63,20 +64,21 @@ public async override Task<ArmOperation<SubnetOperations>> StartCreateAsync(stri
...
             return new PhWrappingAsyncPageable<Subnet, SubnetOperations>(
-                subnetClient.ListAsync(Id.ResourceGroup, Id.Name, cancellationToken),
-                s => new SubnetOperations(ClientContext, new PhSubnet(s, DefaultLocation)));
+                Operations.ListAsync(Id.ResourceGroup, Id.Name, cancellationToken),
+                this.convertor());
         }
+        private Func<Subnet, SubnetOperations> convertor(){
```
Comment (m-nash) [inline azure-proto-network/SubnetContainer.cs:78]: still need to move to newline

File: azure-proto-network/VirtualNetworkContainer.cs [hunk 13]
```diff
@@ -55,20 +57,34 @@ public async override Task<ArmOperation<VirtualNetworkOperations>> StartCreateAs
             return new ArmBuilder<VirtualNetworkOperations, PhVirtualNetwork>(this, new PhVirtualNetwork(vnet));
         }
 
-        public Pageable<VirtualNetworkOperations> List(ArmSubstringFilter filter = null, int? top = null, CancellationToken cancellationToken = default)
+        public Pageable<VirtualNetworkOperations> List(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/VirtualNetworkContainer.cs:60]: still need to move to newline

File: azure-proto-network/VirtualNetworkContainer.cs [hunk 14]
```diff
@@ -55,20 +57,34 @@ public async override Task<ArmOperation<VirtualNetworkOperations>> StartCreateAs
...
+            return new PhWrappingPageable<VirtualNetwork, VirtualNetworkOperations>(
+                Operations.List(Id.Name, cancellationToken),
+                this.convertor());
+        }
+
+        public AsyncPageable<VirtualNetworkOperations> ListAsync(CancellationToken cancellationToken = default){
```
Comment (m-nash) [inline azure-proto-network/VirtualNetworkContainer.cs:66]: still need to move to newline

File: azure-proto-network/VirtualNetworkContainer.cs [hunk 15]
```diff
@@ -55,20 +57,34 @@ public async override Task<ArmOperation<VirtualNetworkOperations>> StartCreateAs
...
 
-        public AsyncPageable<VirtualNetworkOperations> ListAsync(ArmSubstringFilter filter = null, int? top = null, CancellationToken cancellationToken = default)
+        public AsyncPageable<ArmResourceOperations> ListByNameAsync(ArmSubstringFilter filter, int? top = null, CancellationToken cancellationToken = default)
         {
             ArmFilterCollection filters = new ArmFilterCollection(PhVirtualNetwork.ResourceType);
             filters.SubstringFilter = filter;
-            return ResourceListOperations.ListAtContextAsync<VirtualNetworkOperations, PhVirtualNetwork>(ClientContext, Id, filters, top, cancellationToken);
+            return ResourceListOperations.ListAtContextAsync<ArmResourceOperations, ArmResource>(ClientContext, Id, filters, top, cancellationToken);
+        }
+        private  Func<VirtualNetwork, VirtualNetworkOperations> convertor(){
```
Comment (m-nash) [inline azure-proto-network/VirtualNetworkContainer.cs:85]: still need to move to newline

**Review by m-nash — CHANGES_REQUESTED** [review]
the comment about the { on a newline was intended to be general, I went ahead and marked all the occurrences if that makes it easier.

**Review by m-nash — APPROVED** [review]
